pkgs/top-level/all-packages.nix

```diff
@@ -3720,7 +3720,9 @@ with pkgs;
 
   makebootfat = callPackage ../tools/misc/makebootfat { };
 
-  matrix-synapse = callPackage ../servers/matrix-synapse { };
+  matrix-synapse = callPackage ../servers/matrix-synapse {
+    python = python2;
```

Comment:
I'd prefer not having this here but instead use `python2` as an argument.
However, I think that comes down to personal preference.

Reply:
Well, if the point is to allow switching to `python3` and it's almost there, doesn't it make sense to take `python`? Until the switch can be complete and we can take `python3` as an argument :)

Reply:
Alright, why not.

Reply:
One could have inside the expression:
```
let
  python = python2;
in ...
```

The idea is to keep `all-packages.nix` as clean as possible.